### ISSUE TITLE
`QueryResult` is `null` when the `Subscription` widget is built in the first time.

### ISSUE THREAD

**Issue by micimize** [zino-app/graphql-flutter#667]
From https://github.com/zino-app/graphql-flutter/pull/648#issuecomment-640960569

`QueryResult` is `null` when the `Subscription` widget is builded in the first time.

```
════════ Exception caught by widgets library ═══════════════════════════════════════════════════════
The following NoSuchMethodError was thrown building StreamBuilder<QueryResult>(dirty, state: _StreamBuilderBaseState<QueryResult, AsyncSnapshot<QueryResult>>#33392):
The getter 'isLoading' was called on null.
Receiver: null
Tried calling: isLoading

The relevant error-causing widget was: 
  Subscription<dynamic> file:///Users/rodolfosilva/development/sw/lib/ui/screens/overview/overview_tab_screen.dart:22:13
When the exception was thrown, this was the stack: 
#0      Object.noSuchMethod (dart:core-patch/object_patch.dart:53:5)
#1      _OverviewTabScreenState.build.<anonymous closure> (package:sw/ui/screens/overview/overview_tab_screen.dart:27:22)
#2      _SubscriptionState.build.<anonymous closure> (package:graphql_flutter/src/widgets/subscription.dart:174:31)
#3      StreamBuilder.build (package:flutter/src/widgets/async.dart:509:81)
#4      _StreamBuilderBaseState.build (package:flutter/src/widgets/async.dart:127:48)
...
════════════════════════════════════════════════════════════════════════════════════════════════════
```

**Comment by micimize** [comment]
@RodolfoSilva can you provide some the code you're calling `Subscription` with?

**Comment by RodolfoSilva** [comment]
@micimize Yes. I'll post it later ....

**Comment by RodolfoSilva** [comment]
```dart
import 'package:flutter/material.dart';
import 'package:graphql_flutter/graphql_flutter.dart';
import 'package:sw/graphql/graphql_api.dart';

class CustomListView extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    final customersSubscription = CustomersSubscription();

    return Scaffold(
      appBar: AppBar(
        title: Text('Customers'),
      ),
      body: Subscription(
        options: SubscriptionOptions(
          document: customersSubscription.document,
        ),
        builder: (QueryResult result) {
          if (result.isLoading) {
            return Center(
              child: Container(
                child: Column(
                  mainAxisSize: MainAxisSize.min,
                  children: <Widget>[
                    CircularProgressIndicator(),
                    SizedBox(
                      height: 16,
                    ),
                    const Text('Loading...'),
                  ],
                ),
              ),
            );
          }

          final documents =
              customersSubscription.parse(result.data ?? {}).customer;

          if (documents == null || documents.length == 0) {
            return Center(
              child: Text(
                'Oh no!',
                style: TextStyle(fontSize: 24),
              ),
            );
          }

          return ListView.separated(
            itemCount: documents.length,
            separatorBuilder: (_, __) => const Divider(height: 0.0),
            itemBuilder: (context, index) {
              CustomerMixin customer = documents[index];
              return ListTile(
                title: Text(customer.name ?? 'Unknown'),
                subtitle: Text('Phone: ${customer.phone}')
              );
            },
          );
        },
      ),
    );
  }
}
```

**Comment by micimize** [comment]
@RodolfoSilva try `4.0.0-alpha.4`, which adds back an updated version of the previous `WebSocketLink`

My best guess here is that there was some unhandled error in the client, so we need to add `snapshot.error` handling, but I'm still not sure what the best way to do that is.

**Comment by micimize** [comment]
This was actually probably the "initialization from cache" issue solved in [`4.0.0-alpha.6`](https://pub.dev/packages/graphql_flutter/versions/4.0.0-alpha.6/changelog) that was basically just [due to a typo](https://github.com/zino-app/graphql-flutter/commit/de66cffab0c13f1d705bb4538f55a2e11b653a4f)
```diff
      if (cacheResult != null) {
        yield QueryResult(
          source: QueryResultSource.cache,
-          data: options.optimisticResult,
+          data: cacheResult,
        );
      }
```
Closing this for now as I'm fairly sure that's the underlying issue and there hasn't been much activity here – comment, open a new issue, or come to the `#support` channel in [discord](https://discord.gg/tXTtBfC) if the issue persists on `4.0.0-alpha.6`

**Comment by RodolfoSilva** [comment]
Solved, thanks @micimize :P, sorry for my delay.
